google/cloud/spanner/internal/connection_impl.cc

```diff
@@ -1210,8 +1258,18 @@ StatusOr<spanner::CommitResult> ConnectionImpl::CommitImpl(
       break;
     }
     case google::spanner::v1::TransactionSelector::kBegin: {
-      auto begin = BeginTransaction(session, selector->begin(), std::string(),
-                                    ctx, __func__);
+      StatusOr<google::spanner::v1::Transaction> begin;
+      if (session->is_multiplexed()) {
+        // Commit requests containing Mutations on multiplexed sessions require
+        // a random mutation key in order for the service to generate a
+        // precommit token.
+        begin =
+            BeginTransaction(session, selector->begin(), std::string(), ctx,
+                             GetRandomElement(request.mutations()), __func__);
+      } else {
+        begin = BeginTransaction(session, selector->begin(), std::string(), ctx,
+                                 absl::nullopt, __func__);
+      }
```

Comment:
This is another one of those cases where the top-level conditional hides what is conditional and what is fixed.

That is, use `session->is_multiplexed()` to build the optional mutation, and then initialize `begin` with a single `BeginTransaction()` call.